scripts/train_release.js

```diff
@@ -0,0 +1,140 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Script to tag a train across all the repos that need it.
+
+'use strict'
+
+const cp = require('child_process')
+const fs = require('fs')
+const os = require('os')
+const path = require('path')
+
+function err(msg) {
+  console.error()
+  console.error(String(msg))
+  console.error()
+  process.exit(1)
+}
+
+function usage() {
+  err([
+    'Usage: node train_release.js <train version>',
+    '',
+    'Options:',
+    '  --push     Push the tags to the remote repos',
+    '  --verbose  Turn on all stdio'
+  ].join('\n'))
+}
+
+function tempDir(name) {
+  const baseName = `${name}-${Date.now()}-${process.pid}`
+  const dir = path.join(os.tmpdir(), baseName)
+  fs.mkdirSync(dir)
+  return dir
+}
+
+function releaseRepos(repos, version, opts) {
+  const push = !!opts.push
+  const stdio = opts.verbose ? 'inherit' : ['ignore', 'ignore', process.stderr]
+  const tmp = tempDir(`fxa-train-${version}`)
+  console.log(`Temporary dir: ${tmp}`)
+  repos.forEach(repo => {
+    console.log(`Checking ${repo} for ${version}...`)
+    const dir = path.join(tmp, repo)
+    cp.execSync(`git clone git@github.com:mozilla/${repo}`, {
+      cwd: tmp,
+      stdiot: stdio
+    })
+    const cpOpts = {
+      cwd: dir,
+      stdio: stdio
+    }
+    try {
+      cp.execSync(`git checkout ${version}`, {
+        cwd: dir,
+        stdio: 'ignore'
+      })
+      console.log(`Train ${version} already exists for ${repo}, skipping...`)
+
+      console.log(`rm -rf ${repo}`)
+      cp.execSync(`rm -rf ${repo}`, {
+        cwd: tmp,
+        stdio: stdio
+      })
+      return
+    } catch (e) {
+      // tag doesn't exist, creating
+      console.log(`No exising train ${version}, releasing...`)
+    }
+
+    function run(cmd) {
+      console.log(cmd)
+      cp.execSync(cmd, cpOpts)
+    }
+
+    run('rm npm-shrinkwrap.json')
+    run('npm install')
+    run('npm prune')
+    run('npm shrinkwrap --dev')
+    // slice off the v, since grunt version adds it
+    run(`grunt version --setversion=${version.slice(1)}`)
+
+    if (push) {
+      run(`git push origin ${version}`)
+      //run('git push origin master')
+
+      console.log(`rm -rf ${repo}`)
+      cp.execSync(`rm -rf ${repo}`, {
+        cwd: tmp,
+        stdio: stdio
+      })
+    }
+  })
+}
+
+function getTrainVersion(val) {
+  if (!val) {
+    return null
+  }
+  if (!/^[0-9]+$/g.test(val)) {
+    err('Invalid train version: ' + val)
+    return null
+  }
+  let patch = '.0'
+  if (val.indexOf('.') !== -1) {
+    patch = ''
+  }
+  return `v1.${val}${patch}`
+}
+
+function main() {
+  const REPOS = [
+    'fxa-auth-db-mysql',
+    'fxa-auth-mailer',
+    'fxa-auth-server',
+    'fxa-content-server',
+    'fxa-oauth-server',
+    'fxa-profile-server'
+  ]
```

Comment:
> ...it'd be nice if we added the ability to this script to detect if master hasn't changed since its last tag, and so should be tagged at all...

I mostly agree with this but also, I wonder whether there might be hidden benefits to always tagging every release in every repo.

For instance, does it eliminate any apparent confusion that could arise when we've deployed versions with different numbers? Or does nobody ever get confused about that?

What about `/ver.json`, how many repos have it and would it be better if every repo had it and every repo always had the same version number as the wider release?